armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -241,6 +229,41 @@ def construct(
                 runLog.warning(str(e), single=True)
         return b
 
+    def _getBlockwiseMaterialModifierOptions(
+        self, children: Iterable[Composite]
+    ) -> Set[str]:
+        validMatModOptions = set()
+        for c in children:
+            perChildModifiers = self._getCompositeMaterialModifiers(c)
+            validMatModOptions.update(perChildModifiers)
+        return validMatModOptions
+
+    def _getCompositeMaterialModifiers(self, c: Composite) -> Set[str]:
+        perChildModifiers = set()
+        for material in self._getMaterialsInComposite(c):
+            for materialParentClass in material.__class__.__mro__:
+                # we must loop over parents as well, since applyInputParams
+                # could call to Parent.applyInputParams()
+                if issubclass(materialParentClass, Material):
+                    perChildModifiers.update(
+                        signature(
+                            materialParentClass.applyInputParams
+                        ).parameters.keys()
+                    )
+        # self is a parameter to class methods, so it gets picked up here
+        # but that's obviously not a real material modifier
```

Comment:
This. Is. Wild.

Reply:
Yeah it was wild during debugging seeing `self` as modifier collected by looking at the signatures of the various `Material.applyInputParameter` methods